src/components.test.js

```diff
@@ -0,0 +1,105 @@
+import renderer from 'react-test-renderer';
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Calculator from './components/Calculator';
+import App from './App';
+import Quote from './components/Quotes';
+import HomePage from './components/Home';
+
+const Heading = 'Math Magicians';
+const home = 'Math Magicians';
+const quote = 'Mathematics is not about numbers, equations, computations, or algorithems: it is about understanding. -William Paul Thurston';
+const calculatorHeading = 'Lets do some math!';
+
+describe('test rendering', () => {
+  test('Test for page rendering', () => {
+    const tree = renderer.create(<App />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe('navigation test', () => {
+  test('test if home is displayed', () => {
+    render(<App />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(Heading);
+    expect(titles[1]).toHaveTextContent(home);
+    expect(titles.indexOf(quote)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
+
+  test('test if home is displayed', () => {
+    render(<HomePage />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(home);
+    expect(titles.indexOf(quote)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
+
+  test('test if the quotes page works', () => {
+    render(<Quote />);
+    const titles = screen.getAllByRole('heading');
+    expect(titles[0]).toHaveTextContent(quote);
+    expect(titles.indexOf(home)).toBe(-1);
+    expect(titles.indexOf(calculatorHeading)).toBe(-1);
+  });
```

Comment:
- [ ] Good job adding testes to your app , 👍 , most of them are passing as required, but they are still some testes that are not passing , Kindly make sure all of them are passing 👍

List of no passing testes : 
1) Test basic computations › test additions to results div
2) Test basic computations › test additions to results div second try
3) Test basic computations › test basic addition results
4) navigation test › test naviation to calculator

![Capturetest](https://user-images.githubusercontent.com/80458199/220874660-b6ffad29-a8f9-4c0b-a33b-a248f2cc4f42.JPG)

Reply:
is there any way you can approve it. since the changes are optional. because we tried everything possible those tests didn't pass @iambenkis